### PR TITLE
fix: bookmarks ui issues EXLM-1783

### PR DESCRIPTION
**Pull request by nitin-rachabathuni**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-1783

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live/en/actionable-insights?martext=off
- After: https://fix-exlm-1783--exlm--adobe-experience-league.hlx.live/en/actionable-insights?martext=off

### CODE REVIEW

File: scripts/bookmark/bookmark.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { sendNotice } from '../toast/toast.js';
 import { assetInteractionModel } from '../analytics/lib-analytics.js';
```
Comment (noormd08) [inline scripts/bookmark/bookmark.js:2]: Can you please rename this file to bookmark-action.js? Currently we have the block named as bookmark and that lead to confusion.

File: blocks/bookmarks/bookmarks.js [hunk 2]
```diff
@@ -170,11 +205,13 @@ export default async function decorateBlock(block) {
     block.classList.add('bookmarks-empty');
     return;
   }
+  const clonedBookmarkIds = structuredClone(bookmarks);
+  bookmarksEventEmitter.set('bookmark_ids', clonedBookmarkIds);
```
Comment (noormd08) [inline blocks/bookmarks/bookmarks.js:209]: Do we really need a Emitter for this? We need to trigger the profile API and get the latest bookmarks ids (which is integrated in all the places. eg. article-marquee, playlist)

File: scripts/bookmark/bookmark.js (hunk 1 above)
Comment (nitin-rachabathuni) [reply scripts/bookmark/bookmark.js:2]: done Noor.

File: blocks/bookmarks/bookmarks.js (hunk 2 above)
Comment (nitin-rachabathuni) [reply blocks/bookmarks/bookmarks.js:209]: Yes Noor, we have the bookmark' ids fetching API call in a different file and this event happens on a different file, so we are emitting and listening to the event in order to reload bookmarks from the Profile API

**Review by noormd08 — APPROVED** [review]

**Pull request merged**